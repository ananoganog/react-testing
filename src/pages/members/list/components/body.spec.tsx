import React from 'react';
import { shallow } from 'enzyme';
import { Body } from './body';
import { Member } from '../viewModel';


describe('pages/members/list/components spec', () => {
  describe('Body', () => {
    it('should render a body component without rows when passing an empty array of members', () => {
      // Arrange
      const members: Member[] = [];


      // Act
      const component = shallow(<Body members={members} />)

      // Assert
      expect(component.contains('Row')).toBeFalsy();
    });


    it('should render a body component when passing members property', () => {
      // Arrange
      const members: Member[] = [{
        id: 0,
        name: 'someone',
        avatarUrl: 'someone\'s avatar url',
      }];

      // Act
      const component = shallow(<Body members={members} />)

      // Assert
      expect(component).toMatchSnapshot();
    });
  });
});