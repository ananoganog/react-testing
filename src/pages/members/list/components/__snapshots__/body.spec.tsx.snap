// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components spec Body should render a body component when passing members property 1`] = `
<tbody>
  <Component
    key="0"
    member={
      Object {
        "avatarUrl": "someone's avatar url",
        "id": 0,
        "name": "someone",
      }
    }
  />
</tbody>
`;
